api/categorisation.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"context"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular"
+	"github.com/ONSdigital/dp-api-clients-go/v2/cantabularmetadata"
+	"github.com/pkg/errors"
+)
+
+func (api *API) CheckDefaultCategorisation(dimName string, datasetName string) (string, string, error) {
+	ctx := context.Background()
+	labelMap := make(map[string]string)
+	cats, err := api.ctblr.GetCategorisations(ctx, cantabular.GetCategorisationsRequest{
+		Dataset:  datasetName,
+		Variable: dimName,
+	})
+	if err != nil {
+		return "", "", errors.Wrap(err, "failed to check default categorisation")
+	}
+
+	names := make([]string, 0)
+
+	for _, edge := range cats.Dataset.Variables.Edges {
```

Comment:
Unneccesary to be checking the length of these slices/maps before ranging over them

Reply:
The current style is added to make it obvious that we are taking the values from either 'MapFrom' or from 'IsSourceOf' . The query returns both the sections in the result. If this is to be changed, then we will have to check the size of 'names' or 'labelMap' to determine if we have already populated the values and should not try the block starting from line 38.